### ISSUE TITLE
Rapier of lifesteal - the HP gain on hit doesn't work always

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#207]
```
Version 0.6.9

Sometimes the HP gain (0-3 HP) on hit doesn't work when I have the Rapier of 
lifesteal equipped. The gain on kill (3 HP) does always work.

I know there is only a 75% chance that I get 1-3 HP on a hit.

I could reproduce it this way:
1) Load a game (important)
2) hit a monster will always result in a 100% chance of 0 HP gain

I could fix this when doing it this way:
1) Load a game
2) unequip the RoL
3) equip the RoL
4) hit a monster will result in a 75% chance of 1-3 HP gain
```

Original issue reported on code.google.com by `SamuelPl...@gmail.com` on 9 Jul 2011 at 4:49

**Comment by GoogleCodeExporter** [comment]
```
There is something similar happening with all gear, for example if I change an 
item then compile the code uninstall andors trail and install the modified 
client and load a savegame where your character already have the item equiped. 
It wont be changed until you unequip and equip the item again. 

I have not tried to change map, tilt the phone or die after it was done.
```

Original comment by `patri...@hotmail.com` on 10 Jul 2011 at 5:08

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report. 

Yes, this is a bug. The bug was that effects that happen "on hit" are cached on 
the actor whenever those effects change. Of course, they should also be cached 
when the game loads (which they weren't before). This is now fixed so that all 
combat stats are recalculated when a game is loaded.

This fix will also have the added benefit that lingering effects will be 
recalculated when a game loads and the item definition has changed (such as 
when we changed the RoLS to have regeneration).

Thanks for noticing it!

This is now fixed in r122.
```

Original comment by `oskar.wi...@gmail.com` on 13 Jul 2011 at 8:27
- Changed state: **Ready**
- Added labels: **Milestone-0.6.10**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.10 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 21 Oct 2011 at 8:56
- Changed state: **Fixed**
